splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/DecimalIT.java

```diff
@@ -677,4 +684,17 @@ public void testPreparedStatementRetainsFractionalDigits() throws Exception {
                 "14.51000 |";
         testQuery(sqlText, expected, methodWatcher);
     }
+
+    // DB-9425
+    @Test
+    public void testReduceFractionalDigitsForIntegralPart() throws Exception {
+        String sqlText = String.format("select cast(COALESCE(((quantity/units) * ( price / units)* cast(factor AS decimal(10,2)) ),0 ) AS NUMERIC(31,10))\n" +
+                "from DB_9425_TBL --splice-properties useSpark=%s", useSpark);
+
+        String expected = "1              |\n" +
+                "-----------------------------\n" +
+                "3112011525395705.4789182174 |";
+
+        testQuery(sqlText, expected, methodWatcher);
+    }
```

Comment:
i can't see immediately if this covers all the cases above. what i do see is that you don't use + or -, so that code is not tested.
can we have, additional to this, something for sum, for times, for divide separated?

Reply:
When trying to build tests for individual arithmetic operators, I actually found something interesting in our way of deducing division result precision and scale. I document it in the code.

But because of this special thing, I have to follow SQL Server's rules of reducing scale and cannot make it more strict. As a result, we lose scale digits more often and small error accumulates. A lot more tests fail now and I'm trying to get the results from SQL Server for those queries.

Implementation is almost fully changed. Next push would be a force one since delta doesn't make any sense.

Reply:
I end up with implementing SQL Server's scale reduction rules and fixing our result type precision/scale of division. In this way, TPC-H results are correct and only a few tests need to be adapted. Seems to be the best approach so far.